iree/turbine/kernel/wave/scheduling/loop_reconstruction.py

```diff
@@ -99,39 +104,56 @@ def add_nodes_by_schedule(
             # Set the index for the new node by substituting the induction variable
             # for the current iteration.
             new_node.index = node.index
-            for dim in new_node.index:
-                new_node.index[dim] = new_node.index[dim].subs(
-                    {induction_variable: current_induction_variables[iteration]}
-                )
+            if new_node.index:
+                for dim in new_node.index:
+                    new_node.index[dim] = new_node.index[dim].subs(
+                        {induction_variable: current_induction_variables[iteration]}
+                    )
+            if custom_node.expanded_dims:
```

Comment:
Can you explain why this is needed now?

Reply:
The expanded dims are needed for the reshape ops during codegen.